File: Exercises/RB101-RB119_Small_Problems/Easy_7/ex7.rb
# frozen_string_literal: true

# Write a method that takes an Array of integers as input, multiplies all the
# numbers together, divides the result by the number of entries in the Array,
# and then prints the result rounded to 3 decimal places. Assume the array is
# non-empty.

def show_multiplicative_average(arr)
  ave = (arr.reduce(&:*) / arr.size.to_f).round(3).to_s
  # "%.3f" % ave
  format "%.3f", ave
end

puts show_multiplicative_average([3, 5])                # => The result is 7.500
puts show_multiplicative_average([6])                   # => The result is 6.000
puts show_multiplicative_average([2, 5, 7, 11, 13, 17]) # => The result is 28361.667

File: Book_Exercises/IntroToPythonBook/12_Iterators/ex7.py
def find_integers(tup):
    arr = []
    for el in tup:
        if type(el) is int: arr.append(el)
    return arr

my_tuple = (1, 'a', '1', 3, [7], 3.1415,
            -4, None, {1, 2, 3}, False)
integers = find_integers(my_tuple)
print(integers)                    # [1, 3, -4]
